testing/scripts/playwright_test.py

```
@@ -2935,6 +2935,145 @@ async def test_ipv6_mapped6_ui(page: Page) -> None:
+    rows = panel.locator(".zoneid-result__row")
+    # Scheme + Embedded IPv4 + Open-in-tool = 3 rows (no extras for mapped)
+    assert_true("embedded-v4 UI: at least scheme + ipv4 rows",
+                await rows.count() >= 2)
+
+    # Copy embedded IPv4
+    await rows.locator(".subnet-copy").first.click()
+    await page.wait_for_timeout(50)
+    assert_eq("embedded-v4 UI: copy IPv4",
+              await page.evaluate("window.__lastClipboard"), "192.0.2.1")
+
+    # Open-in-tool button is rendered but disabled (per-scheme drawers land in T3–T7)
```

Ruff (0.15.12)

[warning] Comment contains ambiguous `–` (EN DASH). Did you mean `-` (HYPHEN-MINUS)?

(RUF003)